core/trino-main/src/main/java/io/trino/operator/output/Int96PositionsAppender.java

```diff
@@ -98,9 +98,8 @@ public void append(IntArrayList positions, Block block)
     }
 
     @Override
-    public void appendRle(RunLengthEncodedBlock block)
+    public void appendRle(Block block, int rlePositionCount)
```

Comment:
I'm not sure I understand the semantics of this method with the new signature. What's the relationship between `rlePositionCount` and `Block.getPositionCount()`?

What happens if the provided block has more than one position?